### DISCUSSION TITLE
Does helix use a symbol table ? Is it available to view ?

### DISCUSSION THREAD

**Discussion by sritud** [helix-editor/helix#8164] [General]
Does Helix build a symbol table of the variables / functions / arguments or it completely runs on AST generated by tresitter? Can we view the symbol table or the AST  in the editor? 
I am learning compiler design and I would really appreciate if I could be pointed to the right direction. 

Thank you.

**Comment by pascalkuthe** [comment]
Helix is a text editor not a compiler. Language information is provided by language servera. You would need to look at their sourcecode
